test/reactive-handler.spec.ts

```diff
@@ -679,4 +663,139 @@ describe('ReactiveHandler', () => {
             expect(observedTarget).toBe(todos);
         });
     });
+
+    describe('values that do not need to be wrapped', () => {
+        let target;
+        beforeAll(() => {
+            target = new ReactiveMembrane();
+        });
+        it('should not try to make date object reactive', () => {
+            const date = new Date();
+
+            const state = target.getProxy({});
+            state.date = date;
+            expect(state.date).toBe(date);
+        });
+        it('should not try to make inherited object reactive', () => {
+            const foo = Object.create({});
+
+            const state = target.getProxy({});
+            state.foo = foo;
+            expect(state.foo).toBe(foo);
+        });
+        it.each(
+            [undefined, null, 'foo', true, false, 1, Symbol(), () => {}]
+            )('should not treat following value as reactive: %s', (value) => {
+            expect(target.getProxy(value)).toBe(value);
+        })
+    });
+
+    describe('handles array type values', () => {
+        let defaultMembrane;
+        beforeAll(() => {
+            defaultMembrane = new ReactiveMembrane();
+        });
+        it('wraps array', () => {
+            const value = []
+            const proxy = defaultMembrane.getProxy(value);
+            expect(proxy).not.toBe(value);
+            expect(defaultMembrane.unwrapProxy(value)).toBe(value);
+        });
+        it('access items in array', () => {
+            const value = ['foo', 'bar'];
+            const proxy = defaultMembrane.getProxy(value);
+            expect(Array.isArray(proxy)).toBe(true);
+            expect(proxy.length).toBe(2);
+            expect(proxy[0]).toBe('foo');
+            expect(proxy[1]).toBe('bar');
+        });
+        it('should notify when values accessed', () => {
+            const value = ['foo', {nested: 'bar'}];
+            const accessSpy = jest.fn();
+            const observedMembrane = new ReactiveMembrane({
+                valueObserved: accessSpy,
+            });
+    
+            const proxy = observedMembrane.getProxy(value);
+            doNothing(proxy[0]);
+            doNothing(proxy[1]);
+            expect(accessSpy).toHaveBeenCalledTimes(2);
+            expect(accessSpy).toHaveBeenCalledWith(value, '0');
+            expect(accessSpy).toHaveBeenCalledWith(value, '1');
+        });
+
+        describe('access values in array', () => {
+            let membrane;
+            const accessSpy = jest.fn();
+            const changeSpy = jest.fn();
+            beforeAll(() => {
```

Comment:
Why not doing a before `beforeEach` this would avoid having to clear. I think it would make the tests easier to maintain.

```js
let membrane;
let accessSpy;
let changeSpy;

beforeEach(() => {
  accessSpy = jest.fn();
  changeSpy = jest.fn();
  membrane = new ReactiveMembrane({
    valueObserved: accessSpy,
    valueMutated: changeSpy
  });
});
```

Reply:
@pmdartus With the suggested beforeEach() approach, a new membrane and 2 mocks are being created for each testcase in the block. I thought this was not favorable for 2 reasons:
1. It instantiates a new membrane and mock functions for each test case. Why is that required?  The outcome is not dependent on having a fresh membrane or mock for each testcase. 
2. Having many objects managed by one membrane actually is more close to the real world usage of the membrane.

Reply:
This is only my opinion and not some rules that are set in stone =)

> It instantiates a new membrane and mock functions for each test case. Why is that required? The outcome is not dependent on having a fresh membrane or mock for each testcase.

I think each test should be independent if a test, for example, fails it can leave the membrane in an unexpected state. This would have an impact on the outcome of the other tests.

> Having many objects managed by one membrane actually is more close to the real world usage of the membrane.

If you want to test real-world scenario, I would create a test with a new membrane and multiple object test.